cmake/partition_manager.cmake

```diff
@@ -160,7 +163,13 @@ if(PM_IMAGES OR (EXISTS ${static_configuration_file}))
       ${${container}hex_files}
       DEPENDS
       ${${container}targets}
+      # The hex and elf files dependencies both results in a dependency chain
+      # to the child image builds.
+      # Reason for having both dependencies is because:
+      # - hex file: Directly needed for the merging
+      # - elf files: Allows IDE/Debuggers to obtain symbol loading information
       ${${container}hex_files}
+      ${${container}elf_files}
```

Comment:
How does adding this dependency "allow IDE/debuggers to obtain symbol loading information" ?

Intuitively, I would think the hex would suffice as a dependency since the hex "depends" on the elf.

Reply:
no, when the child hex is created as an external project, then the parent image does not know about the elf file itself.